mfs/ops.go

```diff
@@ -3,10 +3,117 @@ package mfs
 import (
 	"errors"
 	"fmt"
+	"os"
+	gopath "path"
 	"strings"
+
+	dag "github.com/ipfs/go-ipfs/merkledag"
 )
 
-func rootLookup(r *Root, path string) (FSNode, error) {
+// Mv moves the file or directory at 'src' to 'dst'
+func Mv(r *Root, src, dst string) error {
+	srcDir, srcFname := gopath.Split(src)
+
+	srcObj, err := Lookup(r, src)
+	if err != nil {
+		return err
+	}
+
+	var dstDirStr string
+	var filename string
+	if dst[len(dst)-1] == '/' {
+		dstDirStr = dst
+		filename = srcFname
+	} else {
+		dstDirStr, filename = gopath.Split(dst)
+	}
+
+	dstDiri, err := Lookup(r, dstDirStr)
+	if err != nil {
+		return err
+	}
+
+	dstDir := dstDiri.(*Directory)
+	nd, err := srcObj.GetNode()
+	if err != nil {
+		return err
+	}
+
+	err = dstDir.AddChild(filename, nd)
+	if err != nil {
+		return err
+	}
+
+	srcDirObji, err := Lookup(r, srcDir)
```

Comment:
there is a race here, and looks like in other places in mfs. grab a lock to make the Mv atomic.